### PR TITLE
Add detailed error message corresponding to the IndexOutOfBoundsException while calling getEntry(...)

### PR DESCRIPTION
**Pull request by seedeed**
Motivation:
`getEntry(...)` may throw an IndexOutOfBoundsException without any error messages.

Modification:

Add detailed error message corresponding to the IndexOutOfBoundsException while calling `getEntry(...)` in HpackDynamicTable.java.

### CODE REVIEW

File: codec-http2/src/main/java/io/netty/handler/codec/http2/HpackDynamicTable.java [hunk 1]
```diff
@@ -83,7 +83,8 @@ public long capacity() {
      */
     public HpackHeaderField getEntry(int index) {
         if (index <= 0 || index > length()) {
-            throw new IndexOutOfBoundsException();
+            throw new IndexOutOfBoundsException("expected: 0 < index(" + index + ") <= length("
+                    + length() + ')');
```
Comment (Bennett-Lynch) [inline codec-http2/src/main/java/io/netty/handler/codec/http2/HpackDynamicTable.java:87]: JDK9 introduced the `Objects.checkIndex​(int index, int length)` method. Consider using the wording consistent with that error message:

https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/19fb8f93c59dfd791f62d41f332db9e306bc1422/src/java.base/share/classes/jdk/internal/util/Preconditions.java#L188-L189

**Comment by normanmaurer** [comment]
@seedeed please let me know once I should re-review.

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by seedeed** [comment]
@normanmaurer PTAL

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Pull request merged**